python/darpinstances/solution_checker.py

```diff
@@ -160,24 +166,52 @@ def check_plan(plan: VehiclePlan, plan_counter: int, instance: DARPInstance, use
 
         equipment = action_data.action.request.equipment
         if equipment != 0:
-            if action_data.action.action_type == ActionType.PICKUP:
+            if is_pickup:
                 if not any(equipment in config for config in available_configurations):
                     print("Equipment {} not available in vehicle equipment list.".format(equipment))
                     plan_ok = False
                 used_equipment.append(equipment)
-            elif action_data.action.action_type == ActionType.DROP_OFF:
+            elif is_drop_off:
                 used_equipment.remove(equipment)
 
         cost += travel_time
 
+        # operation time check
+        start_time = plan.vehicle.operation_start
+        end_time = plan.vehicle.operation_end
+        if (start_time>0 & end_time>0 & (action_data.departure_time < start_time) | (action_data.departure_time > end_time)):
```

Comment:
the operation time should be tested against plan departure and plan arrival times, not against individual actions